allennlp/data/dataset_readers/dataset_utils/ontonotes.py

```diff
@@ -0,0 +1,418 @@
+                # file will be duplicated - one file called filename.gold_skel
+                # and one generated from the preprocessing called filename.gold_conll.
+                if not data_file.endswith("gold_conll"):
+                    continue
+
+                yield os.path.join(root, data_file)
+
+    def sentence_iterator(self, file_path: str) -> Iterator[OntonotesSentence]:
+        """
+        An iterator over the sentences in an individual CONLL formatted file.
+        """
+        with codecs.open(file_path, 'r', encoding='utf8') as open_file:
+            conll_rows = []
+            for line in open_file:
+                line = line.strip()
+                if line != '' and not line.startswith('#'):
+                    conll_rows.append(line)
+                else:
+                    if not conll_rows:
+                        continue
+                    else:
+                        yield self._conll_rows_to_sentence(conll_rows)
+                        conll_rows = []
+
+    def _conll_rows_to_sentence(self, conll_rows: List[str]) -> OntonotesSentence:
+        document_id: str = None
+        sentence_id: int = None
+        # The words in the sentence.
+        sentence: List[str] = []
+        # The pos tags of the words in the sentence.
+        pos_tags: List[str] = []
+        # the pieces of the parse tree.
+        parse_pieces: List[str] = []
+        # The lemmatised form of the words in the sentence which
+        # have SRL or word sense information.
+        predicate_lemmas: List[str] = []
+        # The FrameNet ID of the predicate.
+        predicate_framenet_ids: List[str] = []
+        # The sense of the word, if available.
+        word_senses: List[int] = []
+        # The current speaker, if available.
+        speakers: List[str] = []
+
+        coreference: List[str] = []
+        verbal_predicates: List[str] = []
+        span_labels: List[List[str]] = []
+        current_span_labels: List[str] = []
+
+        # Cluster id -> List of (start_index, end_index) spans.
+        clusters: DefaultDict[int, List[Tuple[int, int]]] = defaultdict(list)
+        # Cluster id -> List of start_indices which are open for this id.
+        coref_stacks: DefaultDict[int, List[int]] = defaultdict(list)
+
+        for index, row in enumerate(conll_rows):
+            conll_components = row.split()
+
+            document_id = conll_components[0]
+            sentence_id = int(conll_components[1])
+            word = conll_components[3]
+            pos_tag = conll_components[4]
+            parse_piece = conll_components[5]
+
+            # Replace brackets in text with a different
+            # token for parse trees.
+            if word == "(":
+                parse_word = "-LRB-"
+            elif word == ")":
+                parse_word = "-RRB-"
+            else:
+                parse_word = word
+            parse_piece = parse_piece.replace("*", f" {parse_word}")
```

Comment:
What does this line do?  Ah, I see you've documented this above; a quick pointer here would be good.  Also, it looks like the docs above say the POS tag should also be in here, right?